analytic_tag_dimension/models/account_analytic_dimension.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2017 PESOL (http://pesol.es) - Angel Moya (angel.moya@pesol.es)
+# Copyright 2020 Tecnativa - Carlos Dauden
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, api, fields, _
+from odoo.exceptions import ValidationError
+
+
+class AccountAnalyticDimension(models.Model):
+    _name = 'account.analytic.dimension'
+    _description = 'Account Analytic Dimension'
+
+    name = fields.Char(required=True)
+    code = fields.Char(required=True)
+    analytic_tag_ids = fields.One2many(
+        comodel_name='account.analytic.tag',
+        inverse_name='analytic_dimension_id',
+        string='Analytic Tags')
+
+    @api.constrains('code')
+    def _check_code(self):
+        for dimension in self:
+            if ' ' in dimension.code:
+                raise ValidationError(_("Code can't contain spaces!"))
+
+    @api.model
+    def get_model_names(self):
+        return [
+            'account.move.line',
+            'account.analytic.line',
+            'account.invoice.line',
+            'account.invoice.report',
+        ]
+
+    def get_field_name(self, code=False):
+        return 'x_dimension_{}'.format(code or self.code).lower()
+
+    @api.model
+    def create(self, values):
+        res = super().create(values)
+        _models = self.env['ir.model'].search([
+            ('model', 'in', self.get_model_names()),
+        ])
+        _models.write({
+            'field_id': [(0, 0, {
+                'name': self.get_field_name(values['code']),
+                'field_description': values.get('name'),
+                'ttype': 'many2one',
```

Comment:
`ttype` is correct field?  It seems incorrect (`type`)

Reply:
Yes, it's correct.